backend/src/api/transactions/transactionRouter.js

```diff
@@ -0,0 +1,28 @@
+const express = require('express');
+const transactionRouter = express.Router();
+const {
+  createTransaction,
+    getTransactionCategories,
+} = require('./transaction.controller');
+const { authorize } = require('../../utils.js/authMiddleware');
+const { validate } = require('../../utils.js/validate');
+const Joi = require('joi');
+
+const addTransactionSchema = Joi.object({
+  amount: Joi.number().required(),
+  transactionDate: Joi.number().required(),
+  type: Joi.string().required(),
+  comment: Joi.string(),
+  category: Joi.string(),
+  userId: Joi.object(),
```

Comment:
Nope, userId should be grabbed from req.user. It's potential security loss, if you allow to create transaction for any user in your application from front-end